service/app/controllers/usermanagement/validator/ShadowUserMigrateReqValidator.java

```diff
@@ -35,12 +37,16 @@ public static  ShadowUserMigrateReqValidator getInstance(Request request,String
     }
 
     public void validate(){
-        checkMandatoryFieldsPresent(request.getRequest(),mandatoryParamsList);
+        validateAction((String)request.getRequest().get(JsonKey.ACTION));
+        if (isActionAccept()) {
```

Comment:
complexity is increase here by adding if and else , we can avoid that like this:
checkMandatoryFieldsPresent (request.getRequest(),isActionAccept()?acceptMandatoryParamsList:rejectMandatoryParamsList);

Reply:
@manzarul sir done